src/tree/updater_quantile_hist.cc

```diff
@@ -19,17 +17,167 @@
 #include "../common/hist_util.h"
 #include "../common/random.h"
 #include "../common/threading_utils.h"
+#include "common_row_partitioner.h"
 #include "constraints.h"
+#include "hist/histogram.h"
 #include "hist/evaluate_splits.h"
 #include "param.h"
 #include "xgboost/logging.h"
 #include "xgboost/tree_updater.h"
 
 namespace xgboost {
+struct RandomReplace {
```

Comment:
Is this used somewhere?

Reply:
Yes. I didn't change this logic yet. So it is used in the [same place](https://github.com/dmlc/xgboost/blob/662289eec9a1b5bc3cb702ab43d38bc470c61a82/src/tree/updater_quantile_hist.cc#L441) as in the master branch

Reply:
> Looks good to me generally, assuming all tests are passing for approx after these changes.

The only failed test is python linting. All tests for 'approx' passed.

Reply:
Please remove it. We will get to that later.

Reply:
Sorry,
your suggestion is not clear for me. The RandomReplce structure is currently used in [QuantileHistMaker::Builder::InitSampling](https://github.com/dmlc/xgboost/blob/662289eec9a1b5bc3cb702ab43d38bc470c61a82/src/tree/updater_quantile_hist.cc#L441). So it can not be deleted without changing the other code.
Can you please clarify your suggestion?

Reply:
I mean only in this pr. Add it later when you need it. Otherwise we are reviewing things where we don't know what they are used for. It's just about keeping your prs small and specific.

Reply:
done. Thanks for your advice.